Opentracing misc

JorikSchellekens opened the pull request:
Removes a redundant user_id tag
Includes the servlet name when tracing one.

richvdh commented on changelog.d/5856.misc:
```diff
@@ -0,0 +1 @@
+Remove user_id opentracing tag and add a servelt_name tag.
```
```suggestion
Remove user_id opentracing tag and add a servlet_name tag.
```

richvdh replied:
I don't see anywhere we remove a `user_id` tag?

richvdh commented on synapse/api/auth.py:
```diff
@@ -209,6 +211,9 @@ def get_user_by_req(
                 request.authenticated_entity = user_id
+                opentracing.set_tag("authenticated_entity", user_id)
+                # there is at least one other place where authenticated entity is
+                # set. user_id is tagged in case authenticated_entity is clobbered
```
`user_id` doesn't seem to be tagged here?

richvdh commented on synapse/api/auth.py:
```diff
@@ -260,6 +265,11 @@ def get_user_by_req(
+            opentracing.set_tag("authenticated_entity", user.to_string())
+            # there is at least one other place where authenticated entity is
+            # set. user_id is tagged incase authenticated_entity is clobbered
+            opentracing.set_tag("user_id", user.to_string())
```
confused. I thought we were removing this?

richvdh commented on synapse/api/auth.py:
```diff
@@ -209,6 +211,9 @@ def get_user_by_req(
             user_id, app_service = yield self._get_appservice_user_id(request)
             if user_id:
                 request.authenticated_entity = user_id
+                opentracing.set_tag("authenticated_entity", user_id)
```
this doesn't seem to be mentioned in the changelog?

JorikSchellekens commented:
Oh, right, I flipped things around in my head, I split the previous pr and made the edits so some of the things I removed now just don't exist.. I see where the confusion about merge order was from.

richvdh commented on synapse/api/auth.py:
```diff
@@ -209,6 +211,9 @@ def get_user_by_req(
             if user_id:
                 request.authenticated_entity = user_id
+                opentracing.set_tag("authenticated_entity", user_id)
+                # there is at least one other place where authenticated entity is
```
I'm not sure this comment adds any value now

JorikSchellekens replied:
True

richvdh replied:
... so can you remove it please? :)

richvdh commented on changelog.d/5856.misc:
```diff
@@ -0,0 +1 @@
+Tag authenticated entity and servlet_name.
```
more context needed here please. Remember this is to be read by random synapse admins in the community...

richvdh replied:
also, could be a `.feature` ?

JorikSchellekens replied:
I didn't know who read the newsfiles. 

The change is so small does it warrant feature? What would constitute as misc?

richvdh replied:
the newsfiles get turned directly into entries in the CHANGES.md file, which then go into the release notes (https://github.com/matrix-org/synapse/releases/tag/v1.3.0 etc).

"misc" is really intended for refactors that have no functional effect.

I'd say nothing is "too small" to be a feature, though it might make sense to group together a bunch of things (eg "Various opentracing-related improvements"), which can be done by giving them the same text in the changelog.d file. 

I think what you've done now is fine though.

JorikSchellekens replied:
Oh, that's cool. I'll keep that in mind

richvdh commented on synapse/api/auth.py:
```diff
@@ -260,6 +265,10 @@ def get_user_by_req(
 
             request.authenticated_entity = user.to_string()
 
```
```suggestion
```

richvdh commented:
btw please remember to squash this when merging, we don't need all 17 commits to change 25 lines...
